src/android/UnityAdsPlugin.java

```diff
@@ -1,99 +1,162 @@
+                Log.d(TAG, String.format("videoAdPlacementId: %s %s", placementId, "onUnityAdsError"));
+                Log.d(TAG, String.format("%s", message));
+                callbackID.error(String.format("[\"%s\",\"%s\"]", message, error.name()));
+            }
+
+            @Override
+            public void onUnityAdsShowStart(String placementId) {
+                Log.v("UnityAdsExample", "onUnityAdsShowStart: " + placementId);
+            }
+
+            @Override
+            public void onUnityAdsShowClick(String placementId) {
+                Log.v(TAG, "CLICKED" + placementId);
                 callbackID.success();
-                return;
             }
 
+            @Override
+            public void onUnityAdsShowComplete(String placementId, UnityAds.UnityAdsShowCompletionState state) {
+                if (state == UnityAds.UnityAdsShowCompletionState.COMPLETED) {
+                    callbackID.success();
+                } else if (state == UnityAds.UnityAdsShowCompletionState.SKIPPED) {
+                    callbackID.error("VIDEO_SKIPPED");
+                } else {
+                    callbackID.error("DID FINISH WITH ERROR");
+                }
+            }
+        };
+
+        private IUnityAdsLoadListener adsLoadListener = new IUnityAdsLoadListener() {
+            @Override
+            public void onUnityAdsAdLoaded(String videoAdPlacementId) {
+                UnityAds.show(getApplicationActivity(), videoAdPlacementId, new UnityAdsShowOptions(), showListener);
+            }
+
+            @Override
+            public void onUnityAdsFailedToLoad(String s, UnityAds.UnityAdsLoadError unityAdsLoadError, String s1) {
+                Log.w(TAG, "onUnityAdsFailedToLoad");
+            }
+        };
+
+        public void show(String videoAdPlacementId) {
+            UnityAds.show(getApplicationActivity(), videoAdPlacementId, new UnityAdsShowOptions(), showListener);
+        }
+
+        private void onInitialized(JSONArray args) {
+            try {
+                String serverId = args.getString(0);
+                if (serverId != null) {
+                    PlayerMetaData playerMetaData = new PlayerMetaData(getApplicationContext());
+                    playerMetaData.setServerId(serverId);
+                    playerMetaData.commit();
+                }
+            } catch (JSONException e) {
+                Log.w(TAG, "serverId arg missing!");
+            }
+
+            String videoAdPlacementId = getVideoPlacementId(args);
+            if (videoAdPlacementId != null) {
+                UnityAds.load(videoAdPlacementId, adsLoadListener);
+            }
+        }
+
+        private void initialize(JSONArray args, CallbackContext callbackContext) {
+            callbackID = callbackContext;
+            this.args = args;
+
             String gameId;
             Boolean testMode = false;
             Boolean debugMode = false;
```

Comment:
Type me be a primitive

```suggestion
            boolean testMode = false;
            boolean debugMode = false;
```